### PR TITLE
Add polygon

### PR DESCRIPTION
**Pull request by linus-sherrill**
Add data structures to support a polygon representation and conversions to other representations.

### CODE REVIEW

File: vital/types/polygon.h [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+   *
+   * This method returns a shared (managed) pointer to the
+   * vital_polygon representation of this abstract polygon. If the
+   * concrete representation is already a vital_polygon, then the
+   * pointer returned refers to this object. If it is of another type,
+   * then the pointer refers to a new object.
+   *
+   * @return Managed pointer to a vital_polygon object.
+   */
+  vital_polygon_sptr get_polygon();
```
Comment (mleotta) [inline vital/types/polygon.h]: I don't see this function implemented anymore, do we still need it?

File: arrows/vxl/tests/test_polygon.cxx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  p->push_back( kwiver::vital::polygon::point_t( 10, 50 ) );
+  p->push_back( kwiver::vital::polygon::point_t( 50, 50 ) );
+  p->push_back( kwiver::vital::polygon::point_t( 30, 30 ) );
+
+  auto xpoly = kwiver::arrows::vxl::vital_to_vxl( p );
+  TEST_EQUAL( "Correct number of vertices, vxl", xpoly->num_vertices(), 4 );
+
+  // Convert back to vital_polygon
+  auto vpoly =  kwiver::arrows::vxl::vxl_to_vital( *xpoly.get() );
+  TEST_EQUAL( "Correct number of vertices, vital", vpoly->num_vertices(), 4 );
```
Comment (mleotta) [inline arrows/vxl/tests/test_polygon.cxx]: might want to also test that the converted values are the same, not just the number of values.

File: vital/types/polygon.h (hunk 1 above)
Comment (linus-sherrill) [inline vital/types/polygon.h]: It looks to be a leftover. Removed.

File: arrows/vxl/tests/test_polygon.cxx (hunk 2 above)
Comment (linus-sherrill) [inline arrows/vxl/tests/test_polygon.cxx]: Added code to test values of vertices

**Pull request merged**